dashboard/src/App.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+import DeploymentOperator from './deployment/DeploymentOperator.js';
+import NoOperator from './NoOperator.js';
+import { apiGet } from './api/api.js';
+import { Container, Segment, Message } from 'semantic-ui-react';
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {operators:{}};
+  }
+
+  async componentDidMount() {
+    this.reloadOperators();
+  }
+
+  async reloadOperators() {
+    const operators = await apiGet('/api/operators');
+    this.setState({operators});
+  }
+
+  render() {
+    setTimeout(this.reloadOperators.bind(this), 5000);
+    let children = [];
+    if (this.state.operators.deployment) {
+      children.push((<DeploymentOperator/>));
+    } else {
+      children.push((<NoOperator/>));
+    }
+    return (
+      <Container>
+        {children}
+        <Segment basic>
+          <Message>
+            <Message.Header>Kube-ArangoDB</Message.Header>
+            <p>
+              Running in Pod <b>{this.state.operators.pod}</b> in namespace <b>{this.state.operators.namespace}</b>.
+            </p>
+          </Message>     
+        </Segment>
+      </Container>
+    );
```

Comment:
TBH, I'd probably extract most of the render function into a separate component that's entirely presentational, e.g.

```jsx
const OperatorsView = ({deployment, pod, namespace}) => (
  <Container>
    {deployment ? <DeploymentOperator /> : <NoOperator />}
    <Segment basic>
      <Message>
        <Message.Header>Kube-ArangoDB</Message.Header>
        <p>Running in Pod <b>{pod}</b> in namespace <b>{namespace}</b>.</p>
      </Message>
    </Segment>
  </Container>
));
class App extends Component {
  state = {}
  componentDidMount () {
    this.reloadOperators();
    this._interval = setInterval(this.reloadOperators, 5000);
  }
  componentWillUnmount () {
    clearInterval(this._interval);
  }
  reloadOperators = async () => {
    const operators = await apiGet('...');
    this.setState({ operators });
  }
  render () {
    if (this.state.operators) {
      return <OperatorsView {...this.state.operators} />;
  }
  return <div>Loading...</div>;
}
```

Keep in mind this still doesn't account for timeouts or errors in the async stuff.

Reply:
FYI the `{...this.state.operators}` part is just me being lazy because I'm coding in the browser. I'd unpack the props explicitly and pass them individually. The `reloadOperators = async () => {...}` stuff binds the function to the instance so you don't need to bind it explicitly when creating the timeout/interval. This is generally a good practice when dealing with things like event handlers but not entirely necessary here (just easier to avoid mistakes by doing it here too).